packages/amplify-category-api/src/provider-utils/awscloudformation/index.ts

```diff
@@ -42,7 +57,171 @@ export async function addResource(context, category, service, options) {
   }
 }
 
-export async function updateResource(context, category, service) {
+export async function addResource(context, category, service, options) {
+  let useContainerResource = false;
+  let apiType = API_TYPE.GRAPHQL;
+
+  if (isAdvanceComputeEnabled(context)) {
+    switch (service) {
+      case 'AppSync':
+        useContainerResource = await isGraphQLContainer(context);
+        apiType = API_TYPE.GRAPHQL;
+        break;
+      case 'API Gateway':
+        useContainerResource = await isRestContainer(context);
+        apiType = API_TYPE.REST;
+        break;
+      default:
+        throw new Error(`${service} not exists`);
+    }
+  }
+
+  return useContainerResource
+    ? addContainerResource(context, category, service, options, apiType)
+    : addNonContainerResource(context, category, service, options);
+}
+
+function isAdvanceComputeEnabled(context) {
+  // TODO: Change this to project setting
+  return FeatureFlags.getBoolean('advancedCompute.enabled');
+}
+
+async function isGraphQLContainer(context): Promise<boolean> {
+  const { graphqlSelection } = await inquirer.prompt({
+    name: 'graphqlSelection',
+    message: 'Which service would you like to use',
```

Comment:
In the flow, this question comes right after "Please select from one of the below mentioned services". It seems wrong to immediately follow that with "Which service would you like to use". I would rename the first service prompt to "Select your API type" (the options are GraphQL and REST), then follow with this question